presto-raptor/src/main/java/com/facebook/presto/raptor/filesystem/RaptorCachingHdfsConfiguration.java

```diff
@@ -50,18 +45,11 @@
     public RaptorCachingHdfsConfiguration(
             @ForCachingFileSystem RaptorHdfsConfiguration hiveHdfsConfiguration,
             CacheConfig cacheConfig,
-            CacheStats cacheStats)
+            CacheManager cacheManager)
     {
         this.hiveHdfsConfiguration = requireNonNull(hiveHdfsConfiguration, "hiveHdfsConfiguration is null");
-
-        CacheConfig config = requireNonNull(cacheConfig, "cachingFileSystemConfig is null");
-        this.cacheManager = config.getBaseDirectory() == null ?
-                new NoOpCacheManager() :
-                new LocalRangeCacheManager(
-                        cacheConfig,
-                        cacheStats,
-                        newScheduledThreadPool(5, daemonThreadsNamed("raptor-cache-flusher-%s")),
-                        newScheduledThreadPool(1, daemonThreadsNamed("raptor-cache-remover-%s")));
+        this.cacheManager = requireNonNull(cacheManager, "CacheManager is null");
+        cacheConfig = requireNonNull(cacheConfig, "cacheConfig is null");
```

Comment:
nit: `this.cacheValidationEnabled = requireNonNull(cacheConfig, "cacheConfig is null").isValidationEnabled();`